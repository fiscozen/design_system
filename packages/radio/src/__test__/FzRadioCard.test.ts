import { mount } from "@vue/test-utils";
import { describe, it, expect, vi, beforeEach } from "vitest";
import { FzRadioCard, FzRadioCardProps } from "..";

const createWrapper = async (props: FzRadioCardProps) => {
  const content = mount(FzRadioCard, {
    props,
  });
  return content;
};

describe("FzRadioCard", () => {
  beforeEach(() => {
    const mockIntersectionObserver = vi.fn();
    mockIntersectionObserver.mockReturnValue({
      observe: () => null,
      unobserve: () => null,
      disconnect: () => null,
    });
    window.IntersectionObserver = mockIntersectionObserver;
  });
  it("should render correctly", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      title: "This is a title",
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should have checked attribute", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      checked: true,
      title: "This is a title",
    });

    expect(wrapper.find("input").element.checked).toBe(true);
  });

  it("should have disabled attribute", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      disabled: true,
      title: "This is a title",
    });

    expect(wrapper.find("input").element.disabled).toBe(true);
  });

  it("should have emphasized class", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      emphasis: true,
      title: "This is a title",
    });

    expect(wrapper.find("label").classes()).toContain(
      "peer-checked:before:border-blue-500",
    );
  });
});
