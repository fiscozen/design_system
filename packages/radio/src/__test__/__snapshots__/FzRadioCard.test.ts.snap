// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzRadioCard > should render correctly 1`] = `
"<div><input type="radio" id="Radio" class="peer h-0 w-0 absolute fz-hidden-input" tabindex="0"><label class="relative flex fz-radio__label block w-[360px] rounded-lg border-solid pt-12 px-12 cursor-pointer pt-[11px] border-2 px-[11px] border-blue-500  bg-grey-hover before:!hidden text-base before:h-16 before:w-16 peer-checked:before:border-[5px] before:border-grey-500 peer-checked:before:border-blue-500" for="Radio">
    <!--v-if-->
    <div class="flex flex-row w-full justify-between">
      <div class="fz-input flex flex-col">
        <p class="font-medium">This is a title</p>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </label></div>"
`;
